integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/DistroXClusterFaultTolerantUpscaleTest.java

```diff
@@ -0,0 +1,233 @@
+                .withImageSettings(DIX_IMG_KEY)
+                .withNetwork(DIX_NET_KEY)
+                .when(distroXClient.create(), key(stack))
+                .await(STACK_AVAILABLE, key(stack).withIgnoredStatues(Set.of(Status.UNREACHABLE)))
+
+                .mockSalt().run().post().bodyContains(saltBodyFilters, 1)
+                .thenReturn(minionIpAddressesResponse, null, 200, 0, null)
+                .when(distroXClient.scale(params.getHostgroup(), SUCCESSFUL_UPSCALE_NODE_COUNT))
+                .awaitForFlow(RunningParameter.waitForFlowFail())
+                .await(STACK_AVAILABLE, key(stack).withPollingInterval(POLLING_INTERVAL))
+                .then((tc, testDto, client) -> verifyHealthyNodeCountAfterUpscale(testDto));
+
+        currentContext.validate();
+    }
+
+    @Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
+    @Description(
+            given = "there is a running DistroX cluster",
+            when = "an upscale is called, but CM's request, waiting for new hosts, times out",
+            then = "the cluster should be AVAILABLE in a partially upscaled state")
+    public void testUpscaleWithWaitingForHostsTimingOut(MockedTestContext testContext, ITestContext testNgContext) {
+        DistroXStartStopTestParameters params = new DistroXStartStopTestParameters(testNgContext.getCurrentXmlTest().getAllParameters());
+        String stack = resourcePropertyProvider().getName();
+        createDatalake(testContext);
+
+        ApiHostList apiHostList = getMockApiHostList();
+
+        DistroXTestDto currentContext = testContext
+                .given(DIX_NET_KEY, DistroXNetworkTestDto.class)
+                .given(DIX_IMG_KEY, DistroXImageTestDto.class)
+                .withImageCatalog()
+                .withImageId(IMAGE_CATALOG_ID)
+                .given(CM_FOR_DISTRO_X, DistroXClouderaManagerTestDto.class)
+                .given(CLUSTER_KEY, DistroXClusterTestDto.class)
+                .withValidateBlueprint(false)
+                .withClouderaManager(CM_FOR_DISTRO_X)
+                .given(stack, DistroXTestDto.class)
+                .withCluster(CLUSTER_KEY)
+                .withName(stack)
+                .withImageSettings(DIX_IMG_KEY)
+                .withNetwork(DIX_NET_KEY)
+                .when(distroXClient.create(), key(stack))
+                .await(STACK_AVAILABLE, key(stack).withIgnoredStatues(Set.of(Status.UNREACHABLE)))
+
+                .mockCm().hosts().get().thenReturn(apiHostList, null, 200, 0, null)
+                .when(distroXClient.scale(params.getHostgroup(), SUCCESSFUL_UPSCALE_NODE_COUNT))
+                .awaitForFlow(RunningParameter.waitForFlowFail())
+                .await(STACK_AVAILABLE, key(stack).withPollingInterval(POLLING_INTERVAL))
+                .then((tc, testDto, client) -> verifyHealthyNodeCountAfterUpscale(testDto));
+
+        currentContext.validate();
+    }
+
+    private MinionIpAddressesResponse getMockResponseForIpAddrsSaltCall() {
```

Comment:
as discussed, in long term we should find out ips and fqdns dinamycally instead of static values, for now this should work since mock will generate same ips and fqdns, but later this should be improved under https://jira.cloudera.com/browse/CB-17921